server/src/main/java/org/elasticsearch/action/ActionModule.java

```diff
@@ -709,6 +710,7 @@ public <Request extends ActionRequest, Response extends ActionResponse> void reg
         actions.register(TransportShardFlushAction.TYPE, TransportShardFlushAction.class);
         actions.register(TransportShardRefreshAction.TYPE, TransportShardRefreshAction.class);
         actions.register(TransportPrevalidateShardPathAction.TYPE, TransportPrevalidateShardPathAction.class);
+        actions.register(TransportUnpromotableShardRefreshAction.TYPE, TransportUnpromotableShardRefreshAction.class);
```

Comment:
Hmm I don't think we should register this here because I see no need for a client to call it directly (unless I'm missing something). Instead, we can register it in the constructor of `TransportShardRefreshAction`.